lgc/state/PipelineState.cpp

```diff
@@ -434,11 +436,8 @@ void PipelineState::record(Module *module) {
   if (m_palMetadata)
     m_palMetadata->record(module);
 
-  if (UseRegisterFieldFormat) {
-    const bool isFieldSupported =
-        getTargetInfo().getGfxIpVersion().major >= 11 && (m_pipelineLink == PipelineLink::WholePipeline);
-    UseRegisterFieldFormat.setValue(isFieldSupported);
-  }
+  if (UseRegisterFieldFormat)
+    UseRegisterFieldFormat.setValue(getTargetInfo().getGfxIpVersion().major >= 11);
```

Comment:
This must have slipped through in an earlier review. Why are we changing the value of a cl::opt here? I think this code should just be removed... (also the setting of UseRegisterFieldFormat in the PipelineState constructor.

Reply:
If the option is turned on and the gfx ip is pre-gfx11, we should fallback to the old mode. 
The reason that we change cl::opt here is to make relocatable shaders work well. In `PalMetadata::record`, we can only check cl::opt `UseRegisterFieldFormat` instead of `m_pipelineState->useRegisterFieldFormat()` to set pal version metaddata since `m_piplineState` is null for relocatable shaders cases.

Reply:
This is about the use from `GlueShader::compile`, right?

Could you please make it so that the `PalMetadata` constructor takes an explicit `bool useRegisterFieldFormat`, which can be programmed by GlueShader because all the cases where a (derived class of) GlueShader is instantiated do have a PipelineState object.

The cleanup could be in a follow-up change if you prefer.

Reply:
Thanks! I will make the cleanup in a follow-up change.